numba_dpex/tests/kernel_tests/test_arg_types.py

```diff
@@ -37,13 +40,39 @@ def input_arrays(request):
 
 @pytest.mark.parametrize("filter_str", filter_strings)
 def test_kernel_arg_types(filter_str, input_arrays):
-    kernel = dpex.kernel(mul_kernel)
-    a, actual, c = input_arrays
+    usm_type = "device"
+
+    a, b, c = input_arrays
     expected = a * c
+
     device = dpctl.SyclDevice(filter_str)
-    with dpctl.device_context(device):
-        kernel[global_size, local_size](a, actual, c)
-    np.testing.assert_allclose(actual, expected, rtol=1e-5, atol=0)
+    queue = dpctl.SyclQueue(device)
+
+    da = dpt.usm_ndarray(
+        a.shape,
+        dtype=a.dtype,
+        buffer=usm_type,
+        buffer_ctor_kwargs={"queue": queue},
+    )
+    da.usm_data.copy_from_host(a.reshape((-1)).view("|u1"))
+
+    db = dpt.usm_ndarray(
+        b.shape,
+        dtype=b.dtype,
+        buffer=usm_type,
+        buffer_ctor_kwargs={"queue": queue},
+    )
+    db.usm_data.copy_from_host(b.reshape((-1)).view("|u1"))
+
+    kernel = dpex.kernel(mul_kernel)
+    kernel[dpex.NdRange(dpex.Range(global_size), dpex.Range(local_size))](
+        da, db, c
```

Comment:
I am surprised that this code works. We are mixing tensor and numpy. @chudur-budur  please check why an exception is not raised here.

`c` should be a numpy array, in fact you can just pass a scalar value. The kernel is only scaling  `a` into `b`.